apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/[ecosystemId]/components/client/ecosystem-header.client.tsx

```diff
@@ -30,54 +29,56 @@ export function EcosystemHeader({ ecosystemId }: { ecosystemId: string }) {
+          <h2 className="text-4xl font-bold text-foreground">
+            {currentEcosystem.name}
+          </h2>
+        )}
+        {!currentEcosystem ? (
+          <Skeleton className="h-5 rounded-md w-[300px]" />
+        ) : (
+          <p className="text-muted-foreground">{currentEcosystem.slug}</p>
+        )}
       </div>
       <div className="flex flex-col items-end justify-between gap-4">
         <Link href="/dashboard/connect/ecosystem/create">
           <Button variant="primary">New Ecosystem</Button>
         </Link>
-        <DropdownMenu>
-          <DropdownMenuTrigger asChild>
-            <Button
-              variant="outline"
-              className="relative flex justify-start w-40 pl-3 pr-8 truncate"
-            >
-              <div className="truncate">{currentEcosystem?.name}</div>
-              <ChevronsUpDown className="absolute w-4 h-4 text-muted-foreground right-2" />
-            </Button>
-          </DropdownMenuTrigger>
-          <DropdownMenuContent className="w-40">
-            {ecosystems?.map((ecosystem) => (
-              <Link
-                href={`/dashboard/connect/ecosystem/${ecosystem.id}`}
-                key={ecosystem.id}
+        {!currentEcosystem ? (
+          <Skeleton className="h-11 rounded-md w-[160px]" />
+        ) : (
+          <DropdownMenu>
+            <DropdownMenuTrigger asChild>
+              <Button
+                variant="outline"
+                className="relative flex justify-start w-40 pl-3 pr-8 truncate"
               >
-                <DropdownMenuItem className="relative flex items-center pl-8 pr-3 cursor-pointer">
-                  {ecosystem.id === ecosystemId && (
-                    <CheckIcon className="absolute w-4 h-4 text-green-500 left-2" />
-                  )}
-                  <div className="truncate">{ecosystem.name}</div>
-                </DropdownMenuItem>
-              </Link>
-            ))}
-          </DropdownMenuContent>
-        </DropdownMenu>
+                <div className="truncate">{currentEcosystem?.name}</div>
+                <ChevronsUpDown className="absolute w-4 h-4 text-muted-foreground right-2" />
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent className="w-40">
+              {ecosystems?.map((ecosystem) => (
+                <Link
+                  href={`/dashboard/connect/ecosystem/${ecosystem.id}`}
+                  key={ecosystem.id}
+                >
+                  <DropdownMenuItem className="relative flex items-center pl-8 pr-3 cursor-pointer">
+                    {ecosystem.id === ecosystemId && (
+                      <CheckIcon className="absolute w-4 h-4 text-green-500 left-2" />
+                    )}
+                    <div className="truncate">{ecosystem.name}</div>
+                  </DropdownMenuItem>
+                </Link>
+              ))}
+            </DropdownMenuContent>
+          </DropdownMenu>
+        )}
```

Comment:
makes me wonder if this shouldn't just be a `<Select>`?

Reply:
Hmm yeah maybe. Select feels a bit weird given they're just links and there's no persistent value to it though

Reply:
well kind of, right? on the "new" page you link to the value of this will be set (based on url)? So it is "persistent" it just is based on url state for the "value"